tensilelite/Tensile/KernelWriterAssembly.py

```diff
@@ -8323,279 +8417,290 @@ def globalWriteElements(self, kernel, tPA, tPB, vectorWidths, elements,
+            #edgeModule.addComment("storeStats, %d, %d, %d"% (edgeI, numSgprs, numElementsPerBatch))
+            # so if we don't have *GPR resources to handle a larger batch then need
+            # to mark overflowedResources rather than generate a kernel that won't work.
+            # Activation
+            actLoopEndLabel, actLoopLabelModules, actLoopEnumStrList = self.initActivationLoop(kernel, beta, edge)
+            actLoopModuleList = []
+            actLoopModuleCodeLength = []
+            with self.allocTmpSgpr(numSgprs, 2) as tmpSgprRes:
+              for index, activationLabelModule in enumerate(actLoopLabelModules):
+                actLoopModule = Module("Activation Loop %s"%index)
+                activationTypeStr = actLoopEnumStrList[index]
+                if activationLabelModule:
+                  actLoopModule.add(activationLabelModule)
+
+                tmpSgpr = tmpSgprRes.idx
+                actTempSgpr = tmpSgpr # Get sgpr start address, should always be the same
+                elementSgprs = tmpSgpr + ss.cfg.numTempSgprPerBatch
+                codeAccVgprRead = deepcopy(self.codes.accVgprRead) if self.states.serializedStore else None
+                mulAlpha = self.codes.mulAlphaMultipleBuffer if kernel["_GlobalAccumulation"] == 'MultipleBuffer' else self.codes.mulAlphaOther
+                codeMulAlpha = deepcopy(mulAlpha) if self.states.serializedStore else None
+
+                self.alphaBeforeLoadC = False
+                if kernel["MIArchVgpr"] and applyAlpha:
+                  codeAccVgprRead = None
+
+                  #Only apply when 2 wave optimization features are enabled
+                  if (kernel["StorePriorityOpt"] or kernel["StoreSyncOpt"]) and beta:
+                    self.alphaBeforeLoadC = True
+                else:
+                  codeMulAlpha = None
+
+                biasLocalBarrierInit = False
+                for batchIdx in range(0, numBatches):
+                  elementStartIdx = batchIdx * numElementsPerBatch
+                  elementStopIdx = min( elementStartIdx + numElementsPerBatch, len(elements[edgeI]) )
+                  elementsThisBatch = elements[edgeI][elementStartIdx:elementStopIdx]
+                  #print("BATCH[%u/%u]: elements[edgeI][%u:%u] VGPRs=%u" % (batchIdx, numBatches, elementStartIdx, elementStopIdx,ss.numVgprsPerElement ))
+                  # elementVgprs can be large and should be perfectly tuned to the number of available
+                  # VGPRS.  We do not want to accidentally overflow and grow the pool here:
+
+                  if kernel["StoreRemapVectorWidth"]:
+                    #Indication if this batch is last batch for this column block shape
+                    self.StoreRemapLastBatch = 1 if (batchIdx+1) % nBatchesPerRow == 0 else 0
+
+                  actLoopModule.add(self.globalWriteBatch(kernel, tPA, tPB, activation, ss, batchIdx, \
```

Comment:
generate global write batch for different bias dim

Reply:
I am thinking if we must separete whole global write into 2 part for BiasM and BiasN.

For Bias read from global into LDS, it can be seperated once before go into global write batch.
For Bias read from LDS, It seems also can be sperated as 2 parts in each batch of global write.

@KKyang Any suggestion on here?

Reply:
Better to seperate them into two kernels because it'll cause the kernel size too large.